src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueueFlushCondition.cs

```diff
@@ -14,6 +14,16 @@ public BlockStoreQueueFlushCondition(IChainState chainState)
         }
 
         /// <inheritdoc/>
-        public bool ShouldFlush { get { return this.chainState.IsAtBestChainTip; } }
+        public bool ShouldFlush
+        {
+            get
+            {
+                // Once store is 5 blocks form the consensus tip then flush on every block.
+                if (this.chainState.ConsensusTip.Height - this.chainState.BlockStoreTip.Height < 5)
```

Comment:
That will flush every block and disable batching. 

So let's say we've started the node and initialized ConsensusTip and BlockStoreTip at height 1000. 
Now CT advances to 1001, BS is still at 1000. 
(1001 - 1000) < 5 == true

I think it worked well like it used to be before.

Reply:
@noescape00 batching will still occur when CT is more than 5 blocks from BST, so to me this change is correct 👍

Reply:
>Batching will basically happen through out IBD, won't it?

Not if internet is too slow or SSD is too fast so consensus can't advance faster than batch can be saved to hard drive. 
In those edge cases we will not have that race condition we rely on and it will result in having a lot of IO calls during IBD. 

>So we now have a situation where batching will happen during IBD 

Most of the times but not always. We rely on race condition for this to happen.

Reply:
Yeah I see the logic in Ivan's claim we will persist to disk in the beginning, so actually we need to also add the IBD check here, I do remember we used to check IBD for some reason it was removed, is the a problem we also check IBD?

Reply:
>for some reason it was removed

Because we go out of IBD a few hundreds blocks before being synced I think

Adding IBD as a condition will help a lot, indeed. 

But still, why do we want this over saving a batch when CT is close to the best peer's tip?

Reply:
We end up with a lot more IO if not.
It was observed in stratis that CT syncs very fast (when blocks are empty) then every block that is pushed to the BSQ async loops is persisted on each dequeue of a block (because CT is now within 5 blocks from best tip) in stratis it can mean more then 2000 blocks.
If anything we should not check CT agasint best peer tip but block BSQ tip to get more correct behaviour.
See my last PR that will address the IBD issue.

Reply:
>It was observed in stratis that CT syncs very fast (when blocks are empty) then every block that is pushed to the BSQ async loops is persisted on each dequeue of a block 

I think this can be fixed just by 
```
public bool ShouldFlush { get { return this.chainState.IsAtBestChainTip && !isIBD; } }
```

Reply:
`isIBD` looks at CT time so it will always be true if `IsAtBestChainTip =true` assuming best tip advances faster then CT.

This will not fix the issue.

Reply:
>isIBD looks at CT time so it will always be true

```
if (this.chainState.ConsensusTip.Header.BlockTime.ToUnixTimeSeconds() < (this.dateTimeProvider.GetTime() - this.consensusSettings.MaxTipAge))
                return true;
```

It won't be true for old blocks, only for those that were created in the last few hours